girder/api/v1/assetstore.py

```diff
@@ -53,33 +52,22 @@ def getAssetstore(self, assetstore, params):
         return assetstore
 
     @access.admin
-    @describeRoute(
+    @autoDescribeRoute(
         Description('List assetstores.')
         .pagingParams(defaultSort='name')
         .errorResponse()
         .errorResponse('You are not an administrator.', 403)
     )
-    def find(self, params):
-        """
-        Get a list of assetstores.
-
-        :param limit: The result set size limit, default=50.
-        :param offset: Offset into the results, default=0.
-        :param sort: The field to sort by, default=name.
-        :param sortdir: 1 for ascending, -1 for descending, default=1.
-        """
-        limit, offset, sort = self.getPagingParameters(params, 'name')
-
-        return list(self.model('assetstore').list(
-            offset=offset, limit=limit, sort=sort))
+    def find(self, limit, offset, sort, params):
+        return list(self.model('assetstore').list(offset=offset, limit=limit, sort=sort))
 
     @access.admin
-    @describeRoute(
+    @autoDescribeRoute(
         Description('Create a new assetstore.')
         .responseClass('Assetstore')
         .notes('You must be an administrator to call this.')
         .param('name', 'Unique name for the assetstore.')
-        .param('type', 'Type of the assetstore.')
+        .param('type', 'Type of the assetstore.', dataType='integer')
```

Comment:
The problem there is that we don't statically know all of the possible assetstore types, some of which may be defined in plugins. I think a future improvement to this infrastructure would be to add some methods to augment existing descriptions via plugins, at which point we could change this to use enum.